diff --git a/node_modules/apify/build/crawlers/basic_crawler.js b/node_modules/apify/build/crawlers/basic_crawler.js
index 3ee6a55..4a12ce4 100644
--- a/node_modules/apify/build/crawlers/basic_crawler.js
+++ b/node_modules/apify/build/crawlers/basic_crawler.js
@@ -272,11 +272,16 @@ class BasicCrawler {
         // Initialize AutoscaledPool before awaiting _loadHandledRequestCount(),
         // so that the caller can get a reference to it before awaiting the promise returned from run()
         // (otherwise there would be no way)
-        this.autoscaledPool = new autoscaled_pool_1.default(this.autoscaledPoolOptions);
-        if (this.useSessionPool) {
-            this.sessionPool = await session_pool_1.openSessionPool(this.sessionPoolOptions);
+        // Making sure, that _init is executed max. once. Otherwise the same event listener would be registered twice in
+        // sessionPool.initialize(), which can lead to corrupted JSON files when persisting the state.
+        if (!this.initCalled) {
+            this.initCalled = true;
+            this.autoscaledPool = new autoscaled_pool_1.default(this.autoscaledPoolOptions);
+            if (this.useSessionPool) {
+                this.sessionPool = await session_pool_1.openSessionPool(this.sessionPoolOptions);
+            }
+            await this._loadHandledRequestCount();
         }
-        await this._loadHandledRequestCount();
     }
     /**
      * @param {object} crawlingContext
